### PR TITLE
Nix dependency updater

### PR DESCRIPTION
**Pull request by vlaci**
Vendored nix dependencies left behind, so I created a friendly action to get those back in order

### CODE REVIEW

File: .github/workflows/update-vendored-nix-dependencies.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        title = f"chore(deps): Updating {path}"
+
+        body = f"""\
+### Changes for {path}
+
+```diff
+{diff}
+```
+"""
```
Comment (qkaiser) [inline .github/workflows/update-vendored-nix-dependencies.py:86]: I would add an explanation about manual action required to merge it.

See below:

```
### Running GitHub Actions on this PR

GitHub Actions will not run workflows on pull requests which are opened by a GitHub Action.

To run GitHub Actions workflows on this PR, run:

```sh
git branch -D update_flake_lock_action
git fetch origin
git checkout update_flake_lock_action
git commit --amend --no-edit
git push origin update_flake_lock_action --force```
```

**Review by qkaiser — COMMENTED** [review]
Cool stuff in a1a7250a48c621335ff3b4b496626368328f54bb !

File: .github/workflows/update-vendored-nix-dependencies.py (hunk 1 above)
Comment (vlaci) [reply .github/workflows/update-vendored-nix-dependencies.py:86]: I am thinking about just creating a repo scoped personal access token TBH, it gets unbearable to manually update everything.

**Comment by vlaci** [comment]
> @vlaci do you know which dependency makes Nix [build Perl](https://github.com/onekey-sec/unblob/actions/runs/5200681263/jobs/9379723231?pr=603#step:7:297) ? I'm confused.

There is a flake.lock update in the branch (while testing the updater, it got there as well). If you check the bottom of the log, it comes through util-linux. Maybe this nix revision is not the best...

File: .github/workflows/update-vendored-nix-dependencies.py (hunk 1 above)
Comment (qkaiser) [reply .github/workflows/update-vendored-nix-dependencies.py:86]: even better

Comment (qkaiser) [reply .github/workflows/update-vendored-nix-dependencies.py:86]: our Sundays will be quieter without the `flake.lock` auto-update merging dance

File: .github/workflows/update-nix.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ jobs:
       - name: Update flake.lock
         uses: DeterminateSystems/update-flake-lock@v19
         with:
+          token: ${{ secrets.CREATE_PR_TOKEN }}
```
Comment (qkaiser) [inline .github/workflows/update-nix.yml:23]: I suppose this token is linked to your account right now. I would rather have it linked to some ONEKEY bot account.

Comment (vlaci) [reply .github/workflows/update-nix.yml:23]: Yep, but at least it is organization managed, so can be revoked and checked from the org dashboard.

There is no "bot" account per se, it'd need to be a regular github account set-up with access to the organization.

Comment (qkaiser) [reply .github/workflows/update-nix.yml:23]: I mean an account like https://github.com/bedevere-bot

Comment (vlaci) [reply .github/workflows/update-nix.yml:23]: https://github.com/onekey-sec/unblob/pull/612

**Review by qkaiser — APPROVED** [review]

**Pull request merged**